crates/bevy_asset/src/lib.rs

```diff
@@ -49,141 +49,122 @@ use bevy_ecs::{
+            processed_file_path: Self::DEFAULT_PROCESSED_FILE_PATH.to_string(),
+            watch_for_changes_override: None,
+        }
     }
 }
 
 impl AssetPlugin {
-    const DEFAULT_FILE_SOURCE: &'static str = "assets";
+    const DEFAULT_UNPROCESSED_FILE_PATH: &'static str = "assets";
     /// NOTE: this is in the Default sub-folder to make this forward compatible with "import profiles"
     /// and to allow us to put the "processor transaction log" at `imported_assets/log`
-    const DEFAULT_FILE_DESTINATION: &'static str = "imported_assets/Default";
-
-    /// Returns the default [`AssetPlugin::Processed`] configuration
-    pub fn processed() -> Self {
-        Self::Processed {
-            destination: Default::default(),
-            watch_for_changes: false,
-        }
-    }
-
-    /// Returns the default [`AssetPlugin::ProcessedDev`] configuration
-    pub fn processed_dev() -> Self {
-        Self::ProcessedDev {
-            source: Default::default(),
-            destination: Default::default(),
-            watch_for_changes: true,
-        }
-    }
-
-    /// Returns the default [`AssetPlugin::Unprocessed`] configuration
-    pub fn unprocessed() -> Self {
-        Self::Unprocessed {
-            source: Default::default(),
-            watch_for_changes: false,
-        }
-    }
-
-    /// Enables watching for changes, which will hot-reload assets when they change.
-    pub fn watch_for_changes(mut self) -> Self {
-        match &mut self {
-            AssetPlugin::Unprocessed {
-                watch_for_changes, ..
-            }
-            | AssetPlugin::Processed {
-                watch_for_changes, ..
-            }
-            | AssetPlugin::ProcessedDev {
-                watch_for_changes, ..
-            } => *watch_for_changes = true,
-        };
-        self
-    }
+    const DEFAULT_PROCESSED_FILE_PATH: &'static str = "imported_assets/Default";
 }
 
 impl Plugin for AssetPlugin {
     fn build(&self, app: &mut App) {
-        app.init_schedule(UpdateAssets)
-            .init_schedule(AssetEvents)
-            .init_resource::<AssetProviders>();
+        app.init_schedule(UpdateAssets).init_schedule(AssetEvents);
+        let embedded = EmbeddedAssetRegistry::default();
         {
-            match self {
-                AssetPlugin::Unprocessed {
-                    source,
-                    watch_for_changes,
-                } => {
-                    let source_reader = app
-                        .world
-                        .resource_mut::<AssetProviders>()
-                        .get_source_reader(source);
-                    app.insert_resource(AssetServer::new(source_reader, *watch_for_changes));
+            let mut sources = app
+                .world
+                .get_resource_or_insert_with::<AssetSourceBuilders>(Default::default);
+            sources.init_default_sources(&self.file_path, &self.processed_file_path);
+            embedded.register_source(&mut sources);
```

Comment:
The "default source" here is a specific thing: the source that paths without sources (ex: `/foo/image.png`) resolve to. Embedded is "added by default" but it is not  "the default source".

This should be renamed to `init_default_source` instead of `init_default_sources` as there is only one.